### PR TITLE
C++ casting and some fixes

### PR DESCRIPTION
**Pull request by lumiZGorlic**
C style casts changed to C++ style casts, also some fixes and cosmetic changes

### CODE REVIEW

**Comment by tigercosmos** [comment]
@lumiZGorlic, thanks for the contribution. Let's fix the CI first. Please also make sure that you have read [the contribution document](https://github.com/seladb/PcapPlusPlus/blob/master/CONTRIBUTING.md).

**Comment by lumiZGorlic** [comment]
Thanks, i read this document before but i got confused about some steps. I just pushed clang-format fixes so hope it'll pass OK now.

File: Packet++/src/SipLayer.cpp [hunk 1]
```diff
@@ -119,8 +119,9 @@ namespace pcpp
 		parseVersion();
 
 		char* endOfFirstLine;
-		if ((endOfFirstLine = static_cast<char*>(memchr(reinterpret_cast<char*>(m_SipRequest->m_Data + m_VersionOffset),'\n',
-		                                                m_SipRequest->m_DataLen - static_cast<size_t>(m_VersionOffset)))) != nullptr)
+		if ((endOfFirstLine =
```
Comment (tigercosmos) [inline Packet++/src/SipLayer.cpp:122]: what's the difference?

**Review by tigercosmos — APPROVED** [review]
LGTM

**Review by seladb — APPROVED** [review]

**Pull request merged**